front/README.md

```diff
@@ -84,35 +84,104 @@ Afterwards:
+[Vitest](https://vitest.dev/) is used for standalone TypeScript functions and modules.
 
-For integration tests, we mock the Apollo client with a `MockedProvider` component instead of the `ApolloProvider` component that is used to handle real data. More information on mocking the Apollo client can be found [here](https://www.apollographql.com/docs/react/development-testing/testing/).
+We use [Mock Service Worker](https://mswjs.io/) (MSW for short) to mock data in a way that faithfully mimics the real server. It's also very useful to mock GraphQL Schema bits that aren't implemented in the production backend yet to build new features on top of it.
 
-To eliminate repetitive code, a custom renderer was built in `front/src/tests/test-utils.js`. It allows developers to render a component in a test environment where chakra, Apollo and Routes are wrapped around it. The utility also exports the entire react testing library, so you should import from this utility instead of `@testing-library/react`. See `EditBox.test.js` for examples of the custom renderer's use.
+Application test specs are inside the `/tests` folder. Other tests are colocated with the files that are the subject of what is being tested.
 
-Tests and test coverage can be run with the following command:
+See Playwright's docs to see how to make the most of its API to help you build test cases e.g. [recording tests by using the app](https://playwright.dev/docs/codegen-intro).
 
-```sh
-# run tests
-docker compose exec front pnpm test
+### Fixtures and mock data
+
+Mock data is located inside `tests/fixtures.ts`, and is usually retrieved from the development backend calls to speed up testing and development since fixtures for that data were already written. As long as both the front-end and backend match the GraphQL Schema, we can be sure that the mocked data will match real data consumed in the app. You can add more fixtures by hand or by using the real API results that you can get from the Network tab inside Devtools in the browser.
+
+To label the fixture/mock data, follow the convention of placing it as `userName: { graphqlOperationName: { modifierEGbaseId: data } }`. Then use it inside the MSW handler at `mswHandlers.ts` (also to export the handler at the bottom of the file).
+
+> [!IMPORTANT]  
+> You must place this entry on your `.env` file for the tests to run against MSW and mock data in `tests/fixtures.ts`.
+>
+> `USE_MSW=true`
+>
+> You might need to bring down the Docker containers, rebuild, and start again to pick up the environment variable changes.
 
-# or locally
+### Running tests
+
+Tests and test coverage can be run with the following commands:
+
+```sh
+# Run tests locally in headless mode (CLI only)
 pnpm test
 
+# Run tests in UI mode (opens a Chromium browser to visually run and debug tests)
+pnpm test:ui
+
 # test coverage
+pnpm test:coverage
+
+# run tests inside Docker
+docker compose exec front pnpm test
```

Comment:
This did not work:
1. no Browsers were installed in the docker container -> add chromium Browser installation to Dockerfile
2. the front docker container is located at /front, but `pnpm test`  only works if you call it from the repo root with `docker compose exec front pnpm -w run test`